Draft: Docker Compose

g0tmi1k opened the pull request.

g0tmi1k committed:
Dockerfile_*: Consistency with ordering

g0tmi1k committed:
Dockerfile_*: Consistency with packages/layers

g0tmi1k committed:
Dockerfile_*: Fix typo auto-apt-proxy

g0tmi1k committed:
Standardized which packages are required to build

This will only do build/compile.
What extra packages are needed to run/function will come later.

g0tmi1k committed:
WIP

g0tmi1k committed:
Standardized which packages are required to build

This will only do build/compile.
What extra packages are needed to run/function will come later.

g0tmi1k committed:
Dockerfiles_*: Update ENTRYPOINT and cargo builds

https://github.com/jelmer/janitor/commit/de675c56d7cbd93c69deebe542918745873516a8

g0tmi1k committed:
Drop python_apt

This was only kept around for brz-debian as it was missing it as dependency (Fixed upstream)

g0tmi1k committed:
Makefile: Break up build & push for docker target

This allows others to build who may not have permission to push back

g0tmi1k committed:
Dockerfiles_*: Use each service in pyproject.toml

g0tmi1k committed:
Dockerfiles_*: Update scripts

g0tmi1k committed:
Makefile: Convert docker/build/push-% to patsubst

Idea is for parallel builds

g0tmi1k committed:
Drop PyPi gpg, Hello python3-gpg

g0tmi1k committed:
Dockerfiles_*: Drop PyPi gpg, Hello python3-gpg

g0tmi1k committed:
Structure - reschedule.py & helpers

REF: https://github.com/jelmer/janitor/pull/935

g0tmi1k committed:
Structure - Containers/Dockerfiles_*

g0tmi1k committed:
Drop Debian Stable

jelmer committed:
Merge pull request #923 from g0tmi1k/dockerfiles

Work on Dockerfiles_*

dependabot[bot] committed:
Bump mailparse from 0.15.0 to 0.16.0

Bumps [mailparse](https://github.com/staktrace/mailparse) from 0.15.0 to 0.16.0.
- [Commits](https://github.com/staktrace/mailparse/compare/v0.15.0...v0.16.0)

---
updated-dependencies:
- dependency-name: mailparse
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #939 from jelmer/1508dependabot/cargo/mailparse-0.16.0

Bump mailparse from 0.15.0 to 0.16.0

jelmer committed:
Merge pull request #929 from g0tmi1k/docs

Work on Docs

g0tmi1k committed:
Site: Add libjs-jquery-datatables

jelmer committed:
Merge pull request #946 from g0tmi1k/libjs

Site: Add libjs-jquery-datatables

g0tmi1k committed:
Fix PSQL for /broken-merge-proposals

jelmer commented on CONTRIBUTING.md:
```diff
@@ -57,3 +57,16 @@ $ sudo apt install \
+Running the containers can be done however is best to suite your environment,
+such as using Docker or Kubernetes.
+
+Example, using Docker:
```
This is using podman rather docker?

g0tmi1k replied:
Yeah, switched :)

jelmer reviewed (changes requested):
This doesn't actually seem to take care of setting up the config and adding volumes for logs and artifacts?

jelmer committed:
Merge pull request #947 from g0tmi1k/broken-merge-proposals

Fix PSQL for /cupboard/broken-merge-proposals

g0tmi1k commented:
Looks of it, also need a psql & redit service running (aka container).

Having a nginx as a reverse proxy in front of all the containers using `--public-port` might be helpful too.

dependabot[bot] committed:
Bump ognibuild from `cecd7d8` to `a93da72`

Bumps [ognibuild](https://github.com/jelmer/ognibuild) from `cecd7d8` to `a93da72`.
- [Release notes](https://github.com/jelmer/ognibuild/releases)
- [Commits](https://github.com/jelmer/ognibuild/compare/cecd7d84196a7dac4320173390dfbcb188f6f1c6...a93da7273c04b2dea5ccf9e54acf45a2ee19c5d9)

---
updated-dependencies:
- dependency-name: ognibuild
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump ruff from 0.9.6 to 0.9.7

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.6 to 0.9.7.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.6...0.9.7)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #953 from jelmer/9b50dependabot/pip/ruff-0.9.7

Bump ruff from 0.9.6 to 0.9.7

jelmer committed:
Merge pull request #952 from jelmer/c74ddependabot/cargo/ognibuild-a93da72

Bump ognibuild from `cecd7d8` to `a93da72`

jelmer committed:
Fix duration test

Fixes #944

jelmer committed:
Merge pull request #954 from jelmer/fix-tests

Fix duration test

jelmer committed:
Format

jelmer committed:
Fix open_branch

jelmer committed:
Fix ruff

jelmer committed:
Fix open_branch reference in janitor.publish

g0tmi1k committed:
Dockerfile_*: Refresh EXPOSE & ENTRYPOINT

g0tmi1k committed:
py/janitor/*.py: Refresh --help

g0tmi1k committed:
py/janitor/*.py: Drop un-used arguments

g0tmi1k committed:
Fix tags

dependabot[bot] committed:
Bump serde from 1.0.217 to 1.0.218

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.217 to 1.0.218.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.217...v1.0.218)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump clap from 4.5.29 to 4.5.31

Bumps [clap](https://github.com/clap-rs/clap) from 4.5.29 to 4.5.31.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v4.5.29...v4.5.31)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump tempfile from 3.16.0 to 3.17.1

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.16.0 to 3.17.1.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.16.0...v3.17.1)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #936 from jelmer/f48ddependabot/cargo/tempfile-3.17.1

Bump tempfile from 3.16.0 to 3.17.1

jelmer committed:
Merge pull request #950 from jelmer/29bedependabot/cargo/clap-4.5.31

Bump clap from 4.5.29 to 4.5.31

jelmer committed:
Merge pull request #951 from jelmer/59c3dependabot/cargo/serde-1.0.218

Bump serde from 1.0.217 to 1.0.218

g0tmi1k committed:
2x new Dockerfiles (auto_upload, mail_filter)

jelmer committed:
Add link to matrix channel

jelmer committed:
Merge pull request #957 from jelmer/matrix-channel

Add link to matrix channel

jelmer committed:
Merge pull request #955 from g0tmi1k/docs

Update Dockerfiles_* EXPOSE & ENTRYPOINT + Cleanup --help

jelmer committed:
Add rust build

jelmer committed:
Revert "Workflow: Containers - Don't push on MR"

We still want to build containers even if we don't push them
(but that was already limited to pushes anyway).

This reverts commit 9b29305152880d1f9dd35ad45f4c3417878d58e6.

dependabot[bot] committed:
Bump ruff from 0.9.7 to 0.9.9

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.7 to 0.9.9.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.7...0.9.9)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump chrono from 0.4.39 to 0.4.40

Bumps [chrono](https://github.com/chronotope/chrono) from 0.4.39 to 0.4.40.
- [Release notes](https://github.com/chronotope/chrono/releases)
- [Changelog](https://github.com/chronotope/chrono/blob/main/CHANGELOG.md)
- [Commits](https://github.com/chronotope/chrono/compare/v0.4.39...v0.4.40)

---
updated-dependencies:
- dependency-name: chrono
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump async-trait from 0.1.86 to 0.1.87

Bumps [async-trait](https://github.com/dtolnay/async-trait) from 0.1.86 to 0.1.87.
- [Release notes](https://github.com/dtolnay/async-trait/releases)
- [Commits](https://github.com/dtolnay/async-trait/compare/0.1.86...0.1.87)

---
updated-dependencies:
- dependency-name: async-trait
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump minijinja from 2.7.0 to 2.8.0

Bumps [minijinja](https://github.com/mitsuhiko/minijinja) from 2.7.0 to 2.8.0.
- [Release notes](https://github.com/mitsuhiko/minijinja/releases)
- [Changelog](https://github.com/mitsuhiko/minijinja/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mitsuhiko/minijinja/compare/2.7.0...2.8.0)

---
updated-dependencies:
- dependency-name: minijinja
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump mailparse from 0.16.0 to 0.16.1

Bumps [mailparse](https://github.com/staktrace/mailparse) from 0.16.0 to 0.16.1.
- [Commits](https://github.com/staktrace/mailparse/compare/v0.16.0...v0.16.1)

---
updated-dependencies:
- dependency-name: mailparse
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #967 from jelmer/1a2ddependabot/cargo/mailparse-0.16.1

Bump mailparse from 0.16.0 to 0.16.1

jelmer committed:
Merge pull request #966 from jelmer/f538dependabot/cargo/minijinja-2.8.0

Bump minijinja from 2.7.0 to 2.8.0

jelmer committed:
Merge pull request #965 from jelmer/48f7dependabot/cargo/async-trait-0.1.87

Bump async-trait from 0.1.86 to 0.1.87

jelmer committed:
Merge pull request #963 from jelmer/c830dependabot/pip/ruff-0.9.9

Bump ruff from 0.9.7 to 0.9.9

jelmer committed:
Merge pull request #964 from jelmer/0cecdependabot/cargo/chrono-0.4.40

Bump chrono from 0.4.39 to 0.4.40

jelmer committed:
Merge pull request #956 from g0tmi1k/new

2x new Dockerfiles(auto_upload, mail_filter)

g0tmi1k committed:
Debian Janitor -> Janitor Bot

g0tmi1k committed:
Runner: new_port -> port

jelmer committed:
Merge pull request #968 from g0tmi1k/debian-janitor

Debian Janitor -> Janitor Bot

g0tmi1k committed:
Rust: Publish - Drop un-used argument

Related commit: e215bbc12c19a555c56ac9fa8083e88dfc77fd95

jelmer committed:
Merge pull request #970 from g0tmi1k/rust

Rust: Publish - Drop un-used argument

jelmer committed:
Merge pull request #971 from jelmer/deploy-containers

Deploy containers

jelmer committed:
Add glossary

jelmer committed:
Drop -Dwarnings for now

jelmer committed:
Don't use pip for building mail filter container

jelmer committed:
Merge pull request #973 from jelmer/glossary

Add glossary

jelmer committed:
Fix tests for Python modules

jelmer committed:
Drop unnecessary steps from python-package.yaml

* rust formatting is checked in rust.yaml
* ognibuild shouldn't need to be built
* 'make all' and 'make' do almost the same thing (minus css)

jelmer committed:
Split out lint workflow with non-Python/rust linting

jelmer committed:
Merge pull request #974 from jelmer/mail-filter-cargo

Don't use pip for building mail filter container

jelmer committed:
Install diffoscope

g0tmi1k committed:
CI: Fix trying to push on pull_request

${{ github.repository_owner }}
  It goes to that new namespace (aka destination) - upstream
  Pull Request (github.event_name: pull_request)

${{ github.actor }}
  The person who open'd up it up (aka source)
  Push Request (github.event_name: push)

g0tmi1k committed:
Site: Debug toolbar (#972)

jelmer committed:
Split out production instructions from CONTRIBUTING

jelmer committed:
Update instructions; apt-based Python packages are explicitly supported

jelmer committed:
Add instructions on running tests

g0tmi1k committed:
Runner: Clean up public port wording (#979)

jelmer committed:
Add more contributing guidelines

jelmer committed:
Merge pull request #980 from jelmer/contributing-guidelines

Contributing guidelines

jelmer committed:
Fix the build (#978)

* Fix ruff issues

* Fix syntax error

jelmer committed:
Add breezy-debian

jelmer committed:
Merge pull request #962 from jelmer/rust-build

Add rust build

jelmer committed:
Split rustfmt check

jelmer committed:
Add janitor to first party

jelmer committed:
Merge pull request #982 from jelmer/first-party

Add janitor to first party

jelmer committed:
Merge pull request #981 from jelmer/rustfmt

Split rustfmt check

jelmer committed:
Convert artifacts module to rust

jelmer committed:
Ignore type error

jelmer committed:
Fix artifacts tests

jelmer committed:
Add rust bindings for running diffoscope

g0tmi1k committed:
Check --config path (#976)

* Check --config path
* Drop un-used --config

jelmer committed:
Handle errors when dpkg-architecture is missing

jelmer committed:
fix formatting

jelmer committed:
Merge pull request #983 from jelmer/artifacts-rust

Convert artifacts module to rust

jelmer committed:
Move debdiff_is_empty to rust

jelmer committed:
Move filter_boring to rust

jelmer committed:
Ignore type errors for artifacts

jelmer committed:
Install diffoscope

g0tmi1k committed:
Fix typos with host/listen_address

g0tmi1k committed:
Worker improvements (#961)

* Worker: Random port if not defined

* Worker: Drop --new-port

* Fix up site_port

---------

Co-authored-by: Jelmer Vernooĳ <jelmer@jelmer.uk>

jelmer committed:
Fix yaml lints (#988)

jelmer committed:
Merge pull request #987 from jelmer/dpkg-architecture-error

Handle errors when dpkg-architecture is missing

jelmer committed:
Merge pull request #985 from jelmer/diffoscope-rust

Add rust bindings for running diffoscope

g0tmi1k committed:
site: Merge plain-layout.html into layout.html

jelmer committed:
Use debsign/dput from silver-platter

jelmer committed:
Add vcs module

jelmer committed:
Wrap vcs manager

jelmer committed:
Merge pull request #969 from g0tmi1k/runner

Runner: new_port -> port

jelmer committed:
Avoid canonicalize until path exists

jelmer committed:
Hopefully fix async tests

jelmer committed:
Fix lint

jelmer committed:
Implement aenter/aexit for backward compatibility

jelmer committed:
Merge pull request #989 from jelmer/fixup

Fixup

dependabot[bot] committed:
Bump ruff from 0.9.9 to 0.9.10

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.9 to 0.9.10.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.9...0.9.10)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #997 from jelmer/dd89dependabot/pip/ruff-0.9.10

Bump ruff from 0.9.9 to 0.9.10

jelmer committed:
Add types

jelmer committed:
Drop removed trace_configs argument. Fixes #990

jelmer committed:
Migrate away from git repositories

jelmer committed:
Hide /git and /bzr if VCS manager is local. Fixes #995

jelmer committed:
Allow unexpected_cfgs

jelmer committed:
More fixes

jelmer committed:
Update debian-control

jelmer committed:
Add notes on reformatting

jelmer committed:
Merge pull request #1003 from jelmer/reformatting

Add notes on reformatting

jelmer committed:
Merge pull request #1001 from jelmer/bugfixes

Several bug fixes

g0tmi1k committed:
janitor -> Janitor

g0tmi1k committed:
Site: Update page_title

g0tmi1k committed:
Janitor Bot -> Janitor

jelmer committed:
Run tests on ubuntu

jelmer committed:
Only provide archive keyring if archive is available

jelmer committed:
Make some targets conditional on publisher or archive being available

jelmer committed:
Add dpkg-dev for dependency as dpkg-dev. Fixes #960

jelmer committed:
Don't pass trace_configs to get_artifact_manager(). Fixes #1009

jelmer committed:
Merge pull request #1005 from g0tmi1k/titles

Site: Update page_title

jelmer committed:
Merge pull request #1004 from g0tmi1k/janitor

janitor( bot)?-> Janitor

jelmer committed:
Merge pull request #992 from g0tmi1k/listen_address

Fix typos with host/listen_address

jelmer committed:
Merge pull request #1010 from jelmer/fix-missing-trace-configs

Fix missing trace configs

jelmer committed:
Merge pull request #1008 from jelmer/worker-dpkg-dev

Add dpkg-dev for dependency dpkg-architecture. Fixes #960

jelmer committed:
Fix flow - remove comment

jelmer committed:
Merge pull request #1013 from jelmer/fixup

Fix flow - remove comment

jelmer committed:
Merge pull request #1007 from jelmer/conditional

Conditionally add publisher/archiver

jelmer committed:
Only setup GPG when necessary

jelmer committed:
Make GPG optional

jelmer committed:
Move uvloop imports to just main

jelmer committed:
Make GPG optional for archive

jelmer committed:
Merge pull request #1014 from jelmer/gpg-opt

Only setup GPG when necessary

jelmer committed:
Make gpg fully optional for archive

g0tmi1k commented:
> This doesn't actually seem to take care of setting up the config and adding volumes for logs and artifacts?

It should now.
Its in a better place, still needs some work

jelmer committed:
Merge pull request #1015 from jelmer/gpg-opt

Make GPG optional in archive

g0tmi1k committed:
Close #1016 - no_pgp -> no_gpg

jelmer committed:
Merge pull request #1017 from g0tmi1k/gpg

Close #1016 - no_pgp -> no_gpg

jelmer committed:
Merge pull request #994 from g0tmi1k/layout

site: Merge plain-layout.html into layout.html

jelmer committed:
Merge pull request #1006 from jelmer/ubuntu-tests

Run tests on ubuntu

g0tmi1k committed:
Fix #1019 - Make sure vcs is defined

g0tmi1k committed:
Update py/janitor/site/simple.py

Co-authored-by: Jelmer Vernooĳ <jelmer@jelmer.uk>

jelmer committed:
Update py/janitor/site/simple.py

jelmer committed:
Update py/janitor/site/simple.py

jelmer committed:
Update py/janitor/site/simple.py

jelmer committed:
Merge pull request #1020 from g0tmi1k/fix-1019

Fix #1019 - Make sure vcs is defined

jelmer committed:
Add python bindings for logs

jelmer committed:
Convert errors

jelmer committed:
Merge pull request #1021 from jelmer/logs-py

Add python bindings for logs

jelmer committed:
Merge pull request #996 from jelmer/debsign-py

Use debsign/dput from silver-platter

jelmer committed:
Add Python bindings for VCS

jelmer committed:
Implement .get_diff_url

jelmer commented:
> It would be 'nice' if its possible to use system variables, rather than config values.
> Otherwise, would need to create another janitor.config just for this (using `example.com` will not work, as it needs containers name)

Not sure I follow

jelmer committed:
More work on vcs python bindings

jelmer committed:
Move djlint config to pyproject

jelmer committed:
Split out html format targets

jelmer committed:
reformat html

jelmer committed:
Fix test

jelmer committed:
Merge pull request #1025 from jelmer/html-reformat

Reformat HTML, and enforce

jelmer committed:
Move get_vcs_managers to rust

jelmer committed:
Drop open_cached_branch, unused

jelmer committed:
Move open_branch_ext to rust

jelmer committed:
Add python bindings for config

jelmer committed:
Use rust config bindings

jelmer committed:
Switch to rust based config

jelmer committed:
Drop Python proto module

jelmer committed:
Fix typing

jelmer commented:
> > Not sure I follow
> 
> With postgresql's container image: https://hub.docker.com/_/postgres, you can set various configs by using system environments (such as `-e POSTGRES_PASSWORD=mysecretpassword`). It would be nice to able todo the same with our container images, and not requiring `janitor.config` file (or the system environment take higher priority).
> 
> i.e.:
> 
>     * `-e JANITOR_DATABASE=https://databfase:5432/janitor`
> 
>     * `-e JANITOR_REDIS=https://redis:6379`

How would that work for e.g. the configuration of the various campaigns?

jelmer committed:
Merge pull request #1024 from jelmer/vcs-py

Add python bindings for vcs interface

g0tmi1k committed:
Fix #1027 - Archive: Import GPG

jelmer committed:
Merge pull request #1028 from g0tmi1k/fix-1027

Fix #1027 - Archive: Import GPG

jelmer committed:
Fix reading from string files

jelmer committed:
Fix ruff

g0tmi1k committed:
CI: Do more than just main branch

REF: https://www.docker.com/blog/docker-best-practices-using-tags-and-labels-to-manage-docker-image-sprawl/

jelmer committed:
Bump pyo3-filelike

jelmer committed:
Wrap section_is_wdiff

jelmer committed:
Merge pull request #1030 from jelmer/fix-ruff

Fix ruff

jelmer committed:
Fix type errors

jelmer committed:
Move run_debdiff to Python

g0tmi1k committed:
Fix #1029 - Runner: Drop trace_configs

jelmer committed:
Move filter_boring_udiff

g0tmi1k committed:
Fix #1011 - Runner: upload_backup_artifacts loop.create_task

dependabot[bot] committed:
Bump tokio-util from 0.7.13 to 0.7.14

Bumps [tokio-util](https://github.com/tokio-rs/tokio) from 0.7.13 to 0.7.14.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-util-0.7.13...tokio-util-0.7.14)

---
updated-dependencies:
- dependency-name: tokio-util
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump async-trait from 0.1.87 to 0.1.88

Bumps [async-trait](https://github.com/dtolnay/async-trait) from 0.1.87 to 0.1.88.
- [Release notes](https://github.com/dtolnay/async-trait/releases)
- [Commits](https://github.com/dtolnay/async-trait/compare/0.1.87...0.1.88)

---
updated-dependencies:
- dependency-name: async-trait
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump http-body-util from 0.1.2 to 0.1.3

Bumps [http-body-util](https://github.com/hyperium/http-body) from 0.1.2 to 0.1.3.
- [Release notes](https://github.com/hyperium/http-body/releases)
- [Commits](https://github.com/hyperium/http-body/compare/http-body-util-v0.1.2...http-body-util-v0.1.3)

---
updated-dependencies:
- dependency-name: http-body-util
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #1037 from jelmer/7aafdependabot/cargo/async-trait-0.1.88

Bump async-trait from 0.1.87 to 0.1.88

jelmer committed:
Merge pull request #1038 from jelmer/65c9dependabot/cargo/http-body-util-0.1.3

Bump http-body-util from 0.1.2 to 0.1.3

jelmer committed:
Fix typing

dependabot[bot] committed:
Bump ruff from 0.9.10 to 0.11.0

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.10 to 0.11.0.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.10...0.11.0)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #1033 from g0tmi1k/fix-1011

Fix #1011 - Runner: upload_backup_artifacts loop.create_task

jelmer committed:
Merge pull request #1032 from g0tmi1k/fix-1029

Fix #1029 - Runner: Drop trace_configs

dependabot[bot] committed:
Bump reqwest from 0.12.12 to 0.12.14

Bumps [reqwest](https://github.com/seanmonstar/reqwest) from 0.12.12 to 0.12.14.
- [Release notes](https://github.com/seanmonstar/reqwest/releases)
- [Changelog](https://github.com/seanmonstar/reqwest/blob/v0.12.14/CHANGELOG.md)
- [Commits](https://github.com/seanmonstar/reqwest/compare/v0.12.12...v0.12.14)

---
updated-dependencies:
- dependency-name: reqwest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump tempfile from 3.18.0 to 3.19.0

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.18.0 to 3.19.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.18.0...v3.19.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #1035 from jelmer/7b6fdependabot/cargo/tempfile-3.19.0

Bump tempfile from 3.18.0 to 3.19.0

jelmer committed:
Merge pull request #1036 from jelmer/3350dependabot/cargo/tokio-util-0.7.14

Bump tokio-util from 0.7.13 to 0.7.14

jelmer committed:
Merge pull request #1039 from jelmer/1373dependabot/pip/ruff-0.11.0

Bump ruff from 0.9.10 to 0.11.0

jelmer committed:
Merge pull request #1040 from jelmer/4ab7dependabot/cargo/reqwest-0.12.14

Bump reqwest from 0.12.12 to 0.12.14

jelmer committed:
Merge pull request #977 from g0tmi1k/ci

CI: Fix trying to push on pull_request

jelmer committed:
Merge pull request #1026 from jelmer/config-py

Use python bindings for config

dependabot[bot] committed:
Bump zip from 2.2.3 to 2.4.1

Bumps [zip](https://github.com/zip-rs/zip2) from 2.2.3 to 2.4.1.
- [Release notes](https://github.com/zip-rs/zip2/releases)
- [Changelog](https://github.com/zip-rs/zip2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zip-rs/zip2/compare/v2.2.3...v2.4.1)

---
updated-dependencies:
- dependency-name: zip
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #1031 from jelmer/debdiff-py

Add python bindings for debdiff

jelmer committed:
Merge pull request #1041 from jelmer/8342dependabot/cargo/zip-2.4.1

Bump zip from 2.2.3 to 2.4.1

jelmer committed:
Replace slashes with dashes in container ref names

jelmer committed:
Merge pull request #1042 from jelmer/replace/slashes

Replace slashes with dashes in container ref names

jelmer committed:
Merge pull request #1043 from jelmer/filter-diff

Move filter_boring_udiff

jelmer committed:
Move get_merged_by_url to rust

jelmer committed:
Migrate branches_match to rust

jelmer committed:
Fix formatting

jelmer committed:
Migrate role_branch_url

jelmer committed:
Fix typing

jelmer committed:
Merge pull request #1045 from jelmer/publish-py

Move some publish functions to python

dependabot[bot] committed:
Bump ruff from 0.11.0 to 0.11.2

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.11.0 to 0.11.2.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.0...0.11.2)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump async-std from 1.13.0 to 1.13.1

Bumps [async-std](https://github.com/async-rs/async-std) from 1.13.0 to 1.13.1.
- [Release notes](https://github.com/async-rs/async-std/releases)
- [Changelog](https://github.com/async-rs/async-std/blob/main/CHANGELOG.md)
- [Commits](https://github.com/async-rs/async-std/compare/v1.13.0...v1.13.1)

---
updated-dependencies:
- dependency-name: async-std
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump log from 0.4.26 to 0.4.27

Bumps [log](https://github.com/rust-lang/log) from 0.4.26 to 0.4.27.
- [Release notes](https://github.com/rust-lang/log/releases)
- [Changelog](https://github.com/rust-lang/log/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rust-lang/log/compare/0.4.26...0.4.27)

---
updated-dependencies:
- dependency-name: log
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump select from 0.6.0 to 0.6.1

Bumps [select](https://github.com/utkarshkukreti/select.rs) from 0.6.0 to 0.6.1.
- [Commits](https://github.com/utkarshkukreti/select.rs/compare/0.6.0...0.6.1)

---
updated-dependencies:
- dependency-name: select
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump silver-platter from 0.5.47 to 0.5.48

Bumps [silver-platter](https://github.com/jelmer/silver-platter) from 0.5.47 to 0.5.48.
- [Release notes](https://github.com/jelmer/silver-platter/releases)
- [Commits](https://github.com/jelmer/silver-platter/commits)

---
updated-dependencies:
- dependency-name: silver-platter
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jelmer committed:
Merge pull request #1046 from jelmer/82f8dependabot/pip/ruff-0.11.2

Bump ruff from 0.11.0 to 0.11.2

jelmer committed:
Merge pull request #1047 from jelmer/8d32dependabot/cargo/async-std-1.13.1

Bump async-std from 1.13.0 to 1.13.1

jelmer committed:
Merge pull request #1050 from jelmer/1adddependabot/cargo/silver-platter-0.5.48

Bump silver-platter from 0.5.47 to 0.5.48

jelmer committed:
Merge pull request #1048 from jelmer/c311dependabot/cargo/log-0.4.27

Bump log from 0.4.26 to 0.4.27

jelmer committed:
Merge pull request #1049 from jelmer/78d5dependabot/cargo/select-0.6.1

Bump select from 0.6.0 to 0.6.1

g0tmi1k committed:
Site: Fix typo

g0tmi1k committed:
Site: Enable --debug

jelmer committed:
Merge pull request #1051 from g0tmi1k/site

Site: Get --debug working

jelmer committed:
Fix type of CampaignConfig.command. Fixes #1054

jelmer committed:
Merge pull request #1056 from jelmer/fix-campaign-config

Fix type of CampaignConfig.command. Fixes #1054

jelmer committed:
Make API more similar to old Python API

jelmer committed:
Fix exception handling

jelmer committed:
Drop use of abi3; prevents subclassing PyException

Since we're on an old PyO3, drop Python 3.13 support for now.

jelmer committed:
Fix formatting

jelmer committed:
Fix typing

jelmer committed:
Merge pull request #1059 from jelmer/rate-limit-py

Convert rate limiters to Rust

g0tmi1k committed:
Close #1062 - Runner: Drop trace_context

jelmer committed:
Merge pull request #1063 from g0tmi1k/trace_context

Runner: Drop trace_context
